dsn.go

```diff
@@ -66,17 +68,48 @@ type Config struct {
 	MultiStatements          bool // Allow multiple statements in one query
 	ParseTime                bool // Parse time values to time.Time
 	RejectReadOnly           bool // Reject read-only connections
+
+	// private fields. new options should be come here
+
+	pubKey       *rsa.PublicKey // Server public key
+	timeTruncate time.Duration  // Truncate time.Time values to the specified duration
 }
 
+// Functional Options Pattern
+// https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis
+type option func(*Config) error
```

Comment:
How about making `option` public?
Users can write code like the following, which I think is useful.

```go
var options []mysql.Option
if some_condition {
  options = append(options, mysql.TimeTruncate(time.Second))
}

cfg := NewConfig(options...)
```